src/docs/guides/phoenix.md

```diff
@@ -0,0 +1,174 @@
+4. **Add a Database Service**:
+    - Run `railway add`.
+    - Select `PostgreSQL` by pressing space and hit **Enter** to add it to your project.
+    - A database service will be added to your Railway project.
+5. **Configure Environment Variables**:
+    - Go to your app service <a href="/overview/the-basics#service-variables">**Variables**</a> section and add the following:
+        - `SECRET_KEY_BASE` : Set the value to the your app's secret key. You can run `mix phx.gen.secret` locally to generate one.
+        - `LANG`and `LC_CTYPE`: Set both values to `en_US.UTF-8`. This sets your app's locale and gets rid of the _native name encoding of latin1_ warning.
+        - `DATABASE_URL`: Set the value to `${{Postgres.DATABASE_URL}}` (this references the URL of your new Postgres database). Learn more about [referencing service variables](/guides/variables#referencing-another-services-variable).  
+        - `ECTO_IPV6`: Set the value to `true` to enable your Phoenix app to connect to the database through the IPv6 private network. 
+    - Use the **Raw Editor** to add any other required environment variables in one go.
+6. **Redeploy the Service**:
+    - Click **Deploy** on the Railway dashboard to apply your changes.
+7. **Verify the Deployment**:
+    - Once the deployment completes, go to **View logs** to check if the server is running successfully.
+8. **Set Up a Public URL**:
+    - Navigate to the **Networking** section under the [Settings](/overview/the-basics#service-settings) tab of your new service.
+    - Click [Generate Domain](/guides/public-networking#railway-provided-domain) to create a public URL for your app.
+
+<Image src="https://res.cloudinary.com/railway/image/upload/f_auto,q_auto/v1730139857/docs/quick-start/phoenix_elixir_app.png"
+alt="screenshot of the deployed Phoenix service showing the welcome page"
+layout="responsive"
+width={2757} height={2111} quality={100} />
+
+## Deploy from a GitHub Repo
+
+To deploy the Phoenix app to Railway, start by pushing the app to a GitHub repo. Once that’s set up, follow the steps below to complete the deployment process.
+
+1. **Create a New Project on Railway**:
+    - Go to <a href="https://railway.app/new" target="_blank">Railway</a> to create a new project.
+2. **Deploy from GitHub**: 
+    - Select **Deploy from GitHub repo** and choose your repository.
+        - If your Railway account isn’t linked to GitHub yet, you’ll be prompted to do so.
+3. **Add Environment Variables**:
+    - Click **Add Variables** and configure all the necessary environment variables for your app.
+        - `SECRET_KEY_BASE` : Set the value to the your app's secret key. You can run `mix phx.gen.secret` locally to generate one.
+        - `LANG`and `LC_CTYPE`: Set both values to `en_US.UTF-8`. This sets our app's locale and gets rid of the _native name encoding of latin1_ warning. 
+4. **Deploy the App**: 
```

Comment:
Can we save this until the last step so we don't need multiple deploys during this?

Reply:
The challenge here is that multiple deploys will still happen regardless. The flow from connecting via GitHub gives an option to add environment variables before deploy so I am taking advantage of that.

However, the app will still fail on initial deploy because there's no Database service yet...which means we still have to provision the DB service, add the env and redeploy.

Reply:
Can you provision the database service in the same step as adding the variables though? so that the database is deployed and the variables are added at the same time.

Reply:
Actually I can 🤔 .  Just tested. The user simply just needs to wait for the DB instance to have deployed (which doesn't take time) before clicking "deploy" on the env variable changes"

Reply:
Fixed.